matpower/matpower-model/src/main/java/com/powsybl/matpower/model/MatpowerBinWriter.java

```diff
@@ -0,0 +1,131 @@
+            gensM.setDouble(row, 11, gens.get(row).getPc2());
+            gensM.setDouble(row, 12, gens.get(row).getQc1Min());
+            gensM.setDouble(row, 13, gens.get(row).getQc1Max());
+            gensM.setDouble(row, 14, gens.get(row).getQc2Min());
+            gensM.setDouble(row, 15, gens.get(row).getQc2Max());
+            gensM.setDouble(row, 16, gens.get(row).getRampAgc());
+            gensM.setDouble(row, 17, gens.get(row).getRampTenMinutes());
+            gensM.setDouble(row, 18, gens.get(row).getRampThirtyMinutes());
+            gensM.setDouble(row, 19, gens.get(row).getRampQ());
+            gensM.setDouble(row, 20, gens.get(row).getApf());
+        }
+
+        List<MBranch> branches = model.getBranches();
+        Matrix branchesM = Mat5.newMatrix(branches.size(), 13);
+        for (int row = 0; row < branches.size(); row++) {
+            branchesM.setDouble(row, 0, branches.get(row).getFrom());
+            branchesM.setDouble(row, 1, branches.get(row).getTo());
+            branchesM.setDouble(row, 2, branches.get(row).getR());
+            branchesM.setDouble(row, 3, branches.get(row).getX());
+            branchesM.setDouble(row, 4, branches.get(row).getB());
+            branchesM.setDouble(row, 5, branches.get(row).getRateA());
+            branchesM.setDouble(row, 6, branches.get(row).getRateB());
+            branchesM.setDouble(row, 7, branches.get(row).getRateC());
+            branchesM.setDouble(row, 8, branches.get(row).getRatio());
+            branchesM.setDouble(row, 9, branches.get(row).getPhaseShiftAngle());
+            branchesM.setDouble(row, 10, branches.get(row).getStatus());
+            branchesM.setDouble(row, 11, branches.get(row).getAngMin());
+            branchesM.setDouble(row, 12, branches.get(row).getAngMax());
+        }
+
+        struct.set("version", Mat5.newString("2"))
+                .set("baseMVA", Mat5.newScalar(100))
+                .set("bus", busesM)
+                .set("gen", gensM)
+                .set("branch", branchesM);
+        return struct;
+    }
+
+    public void write(File oFile) throws IOException {
+        try (Struct struct = fillMatStruct(Mat5.newStruct())) {
+            try (MatFile matFile = Mat5.newMatFile().addArray(MatpowerBinReader.MATPOWER_STRUCT_NAME, struct)) {
+                Mat5.writeToFile(matFile, oFile);
+            }
+        }
+    }
+
+    public void write(OutputStream oStream) throws IOException {
+        try (WritableByteChannel channel = Channels.newChannel(oStream)) {
+            try (Struct struct = fillMatStruct(Mat5.newStruct())) {
+                try (MatFile matFile = Mat5.newMatFile().addArray(MatpowerBinReader.MATPOWER_STRUCT_NAME, struct)) {
+                    ByteBuffer bBuffer = getByteBuffer(matFile);
+                    channel.write(bBuffer);
+                }
+            }
+        }
+    }
+
+    private ByteBuffer getByteBuffer(MatFile matFile) throws IOException {
```

Comment:
Could be static

Reply:
done:  changed this class' methods to static